contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgeable about the project.
+It's OK to ask for help.
+
+There's one key difference though, bringing advantage to you and other people in the future: 
+In almost all cases you should prefer the projects' official communication channels - this can be a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting or the varying needs for structure in the communication.
+All of those usually have in common that they are text-based, archived, searchable and linkable - this means your question and the answer will be written down.
+This way you could benefit from this passively documented knowledge on your search and will help future contributors to have the same advantage.
+Imagine what would be needed for realizing this benefit by just walking by or calling a trusted committer on the project or even in a group call.    
+
+Did you on your way of trying to unblock yourself find that you were missing documentation or found parts of it to be wrong or out-of-date and had to figure this out for yourself now?
+Often projects are happy to receive additions, updates or corrections for their existing documentation as well - you've just found another opportunity to contribute! 
+(Or just politely provide them with feedback on your experience and what would have helped you.)
+
+### Crafting the code
+
+We all have our preferences and opinions on code style, indentation, etc. 
+The home that the host team lives in will have them as well, individual wallpaper, decoration and hangings incl. 
+Try to adapt and match those preferences even if it's not what you would normally do and even if it is not specified in the projects' `CONTRIBUTING.md`.
+If you are unsure you can always ask politely, nevertheless a guest contribution for a feature or bug fix is not the time to introduce a new way of structuring or formatting project code.
+
+## Submitting the pull request
+
+You've completed all the essential work, figured out all the quirks of the problem and the project you are contributing to, the time you've planned for the new feature to be used comes nearer and you want to make sure your contribution gets merged in as fast and smooth as possible?
+
+Here's what you can do to make reviewing and merging as easy as possible for the the trusted committer and the host team.
+(This might actually be pretty similar to what you might already be doing on your own project to get your changes accepted - if that's the case: great, this is going to come natural to you!)
+
+### Testing and automation
+
+The basic point here is to enable the trusted committer to validate the contribution without your presence and to ensure easy maintainability.
```

Comment:
If you need a real story for that:

I once added a feature that I needed at work to an open source project where I was committer - only to have it removed in the next performance optimzation round. It didn't come with a test, so on removal nobody was notified that it was actually needed - nor what it would have been needed for.

Reply:
Valid point, do you see a point where this example could be added while keeping the reading flow? 
I tried and couldn't really decide upon a place to put it.
Would be very about a commit suggestion here.

Reply:
Added a short commit suggestion incorporating your story. Take a look @MaineC 

```suggestion
The basic point here is to enable the trusted committer to validate the contribution without your presence and to ensure easy maintainability.
Imagine you've built a feature or handling of an unsolvable quirk or important performance tweak whose code is not entirely obvious or might even look hacky / wrong at the first glance. 
If you have covered this with a test - and ideally have shed some words on the rationale behind it in a comment - a future editor will get reminded about the purpose of the code and the test(s) will ensure that the value your code realizes will be kept, even in new implementations.
```

Reply:
Offline comment from @MaineC: wfm